### PR TITLE
#182631478 - USER SHOULD BE ABLE TO COMMENT ON AN ARTICLE

### PR DESCRIPTION
**Pull request by dalienst**
#### What does this Pull Request do?

- Enables a user to comment on an article

#### Description of Task to be completed?

- POST: api/articles/comments/
- GET: api/articles/comments/
- DELETE: api/articles/comments/<str:comment_id>/

#### How should this be manually tested?

- Using postman test every endpoint above

#### Any background context you want to provide?

- Ability of user to comment on an article

#### What are the relevant pivotal tracker?

- [182631478](https://www.pivotaltracker.com/n/projects/2570600/stories/182631478)

#### Screenshots (if appropriate)

Screenshot from 2022-08-05 09-56-42

Screenshot from 2022-08-05 09-53-30

Screenshot from 2022-08-05 10-03-26

#### Questions:

- N/A

### CODE REVIEW

File: app/articles/serializers.py [hunk 1]
```diff
@@ -81,3 +83,54 @@ def create(self, validated_data: Any) -> Any:
...
+        model = ArticleComment
+        fields = (
+            "comment_id",
+            "commenter",
+            "comment",
+            "article",
+        )
+        read_only_fields = (
+            "created_at",
+            "commenter",
```
Comment (mainamuriuki274) [inline app/articles/serializers.py:128]: Add comment_id to read only fields,

File: app/articles/tests/test_views.py [hunk 2]
```diff
@@ -108,3 +108,145 @@ def test_delete_article_without_authentication(self) -> None:
...
+        )
+        token = json.loads(response.content).get("access")
+        return {"HTTP_AUTHORIZATION": f"Bearer {token}"}
+
+    def test_bookmark_article(self) -> None:
+        """
+        Test if a user can bookmark an article
+        """
+        data = Article.objects.create(
+            post_id="9831040d-5eeb-4cd5-9054-d0f1770431ca",
```
Comment (mainamuriuki274) [inline app/articles/tests/test_views.py:144]: stop manually putting the id

File: app/articles/tests/test_views.py [hunk 3]
```diff
@@ -108,3 +108,145 @@ def test_delete_article_without_authentication(self) -> None:
...
+        res_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(res_data["article"], data.post_id)
+
+    def test_get_bookmark(self) -> None:
+        """
+        Test if user can get their bookmarks
+        """
+        data = Article.objects.create(
+            post_id="9831040d-5eeb-4cd5-9054-d0f1770431ca",
```
Comment (mainamuriuki274) [inline app/articles/tests/test_views.py:164]: Stop manually putitng the id

File: app/articles/views.py [hunk 4]
```diff
@@ -38,3 +47,38 @@ def delete(self, request: Request, *args: Any, **kwargs: Any) -> Response:
...
+        return queryset.filter(user=self.request.user)
+
+
+class ArticleCommentView(generics.ListCreateAPIView):
+    serializer_class = ArticleCommentSerializer
+    queryset = ArticleComment.objects.all()
+    permission_classes = [IsAuthenticated]
+
+
+class ArticleCommentDeleteView(generics.RetrieveDestroyAPIView):
```
Comment (mainamuriuki274) [inline app/articles/views.py:70]: Change name

**Review by mainamuriuki274 — CHANGES_REQUESTED** [review]

File: app/articles/tests/test_views.py (hunk 2 above)
Comment (dalienst) [reply app/articles/tests/test_views.py:144]: I tried calling it automatically it keeps giving me an assertion error.

File: app/articles/serializers.py [hunk 5]
```diff
@@ -136,6 +143,33 @@ def create(self, validated_data: Any) -> Any:
...
+    """
+    Comment and highlighting serializer
+    """
+
+    commenter = UserSerializer(read_only=True)
+
+    class Meta:
+        model = ArticleComment
+        fields = (
+            "comment_id",
```
Comment (mainamuriuki274) [inline app/articles/serializers.py:156]: add this as a read only field

File: app/articles/urls.py [hunk 6]
```diff
@@ -15,5 +17,11 @@
         name="article-detail",
     ),
     path("bookmarks/", ArticleBookmarkView.as_view(), name="bookmark"),
+    path("comments/", ArticleCommentView.as_view(), name="comment"),
+    path(
+        "comments/<str:comment_id>/",
+        ArticleCommentDeleteView.as_view(),
```
Comment (mainamuriuki274) [inline app/articles/urls.py:23]: Change name to be ArticleDetailView

**Review by mainamuriuki274 — CHANGES_REQUESTED** [review]

File: app/articles/urls.py (hunk 6 above)
Comment (dalienst) [reply app/articles/urls.py:23]: Done

**Review by mainamuriuki274 — APPROVED** [review]

**Pull request merged**